mslib/msui/mpl_pathinteractor.py

```diff
@@ -354,62 +355,31 @@ class PathInteractor(QtCore.QObject):
 
     # picking points
 
-    def __init__(self, ax=None, waypoints=None, mplpath=None,
-                 facecolor='blue', edgecolor='yellow',
-                 linecolor='blue', markerfacecolor='red',
-                 marker='o', label_waypoints=True):
-        """The constructor initializes the path patches, overlying line
-           plot and connects matplotlib signals.
-
-        Arguments:
-        ax -- matplotlib.Axes object into which the path should be drawn.
-        waypoints -- flighttrack.WaypointsModel instance.
-        mplpath -- matplotlib.path.Path instance
-        facecolor -- facecolor of the patch
-        edgecolor -- edgecolor of the patch
-        linecolor -- color of the line plotted above the patch edges
-        markerfacecolor -- color of the markers that represent the waypoints
-        marker -- symbol of the markers that represent the waypoints, see
-                  matplotlib plot() or scatter() routines for more information.
-        label_waypoints -- put labels with the waypoint numbers on the waypoints.
-        """
+    def __init__(self, myfig):
+        self.myfig = myfig
         QtCore.QObject.__init__(self)
         self.waypoints_model = None
         self.background = None
         self._ind = None  # the active vertex
 
         # Create a PathPatch representing the interactively editable path
         # (vertical profile or horizontal flight track in subclasses).
-        path = mplpath
-        pathpatch = mpatches.PathPatch(path, facecolor=facecolor,
-                                       edgecolor=edgecolor, alpha=0.15)
-        ax.add_patch(pathpatch)
+        path = self.myfig.mplpath
+        pathpatch = mpatches.PathPatch(path, facecolor=self.myfig.facecolor,
+                                       edgecolor=self.myfig.edgecolor, alpha=0.15)
+        self.myfig.ax.add_patch(pathpatch)
 
-        self.ax = ax
-        self.path = path
         self.pathpatch = pathpatch
```

Comment:
OK, had a closer look. This is a mess. There are *two* related objects, the pathpatch (which is not plotted) and the line (which is plotted). Both should be identical and the pathpatch is used to interact with the "path", whereas the line is plotting it.
This might be very useful for your disentanglement, but ultimately, it seems overcomplicated and superfluous.
To get the pathpatch to be plotted, it should be sufficient to set "edgecolor" to "linecolor" (and remove the self.line-plot). This works reasonably well - except for drag-dropping points. The latter could likely be fixed in some fashion, probably, by exchanging the line.set_data-stuff with something similar for the pathpatch.
The whole thing is a tad complicated also due to the distinction between wp_vertices and vertices etc needed by the great-circle paths...